xarray/core/computation.py

```diff
@@ -926,6 +926,103 @@ def earth_mover_distance(first_samples,
         return apply_array_ufunc(func, *args, dask=dask)
 
 
+def dot(*arrays, **kwargs):
+    """ dot(*arrays, *, dims=None)
+
+    einsum for xarray object, but providing simpler interface based on
+    the array dimensions.
```

Comment:
We should lead with a more general description. Maybe:

    Generalized dot product for xarray objects. Like np.einsum, but 
    provides a simpler interface based on array dimensions.